packages/back-end/src/api/features/postFeature.ts

```diff
@@ -0,0 +1,157 @@
+    );
+  }
+};
+
+export const validateDefaultValueType = (
+  valueType: z.infer<typeof postFeatureValidator.bodySchema>["valueType"],
+  defaultValue: z.infer<typeof postFeatureValidator.bodySchema>["defaultValue"]
+) => {
+  const defaultValueType = typeof defaultValue;
+  switch (valueType) {
+    case "string":
+    case "number":
+    case "boolean":
+      if (defaultValueType !== valueType)
+        throw new Error(
+          `Type mismatch between valueType ('${valueType}') and defaultValue ('${defaultValueType}').`
+        );
+      break;
+    case "json":
+      if (defaultValueType === "object") return;
+      if (defaultValueType === "string") {
+        try {
+          const json = JSON.parse(defaultValue);
+          // check against valid json that doesn't produce an object
+          // e.g. JSON.parse('false')
+          if (typeof json !== "object") throw new Error("not json obj");
+        } catch (e) {
+          throw new Error(`defaultValue is not valid JSON.`);
+        }
+      } else {
+        throw new Error(
+          "Values of type 'json' must be provided in string or object format."
+        );
+      }
+      break;
+    default:
+      throw new Error(`Invalid valueType ('${valueType}').`);
+  }
+};
+
+export const postFeature = createApiRequestHandler(postFeatureValidator)(
+  async (req): Promise<PostFeatureResponse> => {
+    req.checkPermissions("manageFeatures", req.body.project);
+
+    const existing = await getFeature(req.organization.id, req.body.id);
+    if (existing) {
+      throw new Error(`Feature id '${req.body.id}' already exists.`);
+    }
+
+    const orgEnvs = req.organization.settings?.environments || [];
+
+    // ensure environment keys are valid
+    validateEnvKeys(
+      orgEnvs.map((e) => e.id),
+      Object.keys(req.body.environments ?? {})
+    );
+
+    // ensure default value matches value type
+    validateDefaultValueType(req.body.valueType, req.body.defaultValue);
+
+    const environmentSettings = createInterfaceEnvSettingsFromApiEnvSettings(
+      orgEnvs,
+      req.body.environments ?? {}
+    );
+
+    const defaultValue =
+      req.body.valueType === "json" && typeof req.body.defaultValue === "object"
+        ? JSON.stringify(req.body.defaultValue)
+        : req.body.defaultValue ?? "";
+
+    const feature: FeatureInterface = {
+      defaultValue,
+      valueType: req.body.valueType,
+      owner: req.body.owner,
+      description: req.body.description || "",
+      project: req.body.project || "",
+      dateCreated: new Date(),
+      dateUpdated: new Date(),
+      organization: req.organization.id,
+      id: req.body.id.toLowerCase(),
+      archived: !!req.body.archived,
+      revision: {
+        version: 1,
+        comment: "New feature",
+        date: new Date(),
+        publishedBy: {
+          id: req.body.owner,
+          email: req.body.owner,
+          name: req.body.owner,
+        },
+      },
+      jsonSchema: {
+        schema: "",
```

Comment:
- Accept jsonSchema input
- Set if org is enterprise
- Ignore if not
- Mention in docs